lib/nats/io/client.rb

```diff
@@ -332,6 +355,18 @@ class << self; alias_method :request, :old_request; end
 
       validate_settings!
 
+      self
+    end
+
+    # Establishes a connection to NATS immediately.
+    def connect!(uri = nil, opts = {})
+      connect(uri, opts)
+      establish_connection!
+    end
+
+    private def establish_connection!
```

Comment:
For now I extracted this method and marked it as private in place to make diff smaller.
Probably it should go somewhere below, or maybe not.